### ISSUE TITLE
ProxmoxMachine objects are not finalized

### ISSUE THREAD

**Issue by natitomattis** [ionos-cloud/cluster-api-provider-proxmox#265]
**What steps did you take and what happened:**
When deleting a machine, the Proxmox machine object is not terminated properly. I see the following errors in the controller 
```
Reconciler error" err="cannot delete vm with id 102: 500 Internal Server Error" controller="proxmoxmachine" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="ProxmoxMachine" ProxmoxMachine="backend-at1-dev/backend-at1-dev-accelerated-847gn" namespace="backend-at1-dev" name="backend-at1-dev-accelerated-847gn" reconcileID="934e466a-aef9-4467-b0b2-35a3ec3a043b"
```
It seems like the machine was removed in a previous call and now the controller is failing to find it. 

**What did you expect to happen:**
ProxmoxMachine should be removed

**Anything else you would like to add:**
- I think the issue is related to [this one](https://github.com/ionos-cloud/cluster-api-provider-proxmox/issues/96)
- I found that the error returned on [this line](https://github.com/ionos-cloud/cluster-api-provider-proxmox/blob/main/pkg/proxmox/goproxmox/api_client.go#L154) is not the expected `does not exist`. The error returned when the VM ID does not exist is `500 Internal Server Error`

**Environment:**

- Cluster-api-provider-proxmox version: 0.5.1
- Kubernetes version: (use `kubectl version`): v1.28.11 
- OS (e.g. from `/etc/os-release`): Ubuntu 24

**Comment by mcbenjemaa** [comment]
Thanks for reporting this, we will check this.

**Comment by mcbenjemaa** [comment]
We have merged #278. I'm hoping this will fix the issue.
In my tests, I haven't encountered this issue after merging.
